packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2254,20 +2218,25 @@ export class ContainerRuntime
 					this.idCompressor !== undefined,
 					0x67c /* IdCompressor should be defined if enabled */,
 				);
-				this.idCompressor.finalizeCreationRange(message.contents as IdCreationRange);
+				this.idCompressor.finalizeCreationRange(message.contents);
```

Comment:
My IDE is showing `message.contents: any` here.  I think this code should be operating under the assumption that the thing is a Runtime thing.  `|`ing in `ISequencedDocumentMessage` is watering this down I think

Reply:
I am seeing that too.
Even if I temporarily move the `!messageWithContext.modernRuntimeMessage` check to top to narrow to `InboundSequencedContainerRuntimeMessage` it doesn't get to just `ISequencedDocumentMessage & InTransitContainerRuntimeIdAllocationMessage`
(like it gets to `ISequencedDocumentMessage & UnknownContainerRuntimeMessage` for the default case).

We need `ISequencedDocumentMessage['contents']` to be `unknown` instead of `any` to get the narrowing. But even then that is only if the `modernRuntimeMessage` check is performed, which I think is a no go.

So we need something that better. Probably a type that omits `type` and/or at least  `contents` from `ISequencedDocumentMessage`. Thinking...

Reply:
I vote to leave it for now and try a scoped follow-up PR to improve it.

Reply:
The problem is that it isn't an improvement. We should at least restore the as.